GHzDACs/Cleanup/dac.py

```diff
@@ -936,8 +1024,534 @@ class DacRunner_Build11(DacRunner_Build7):
+        regs[45] = sync
+        regs[17] = 0  # Which jump table to count activations of
+        regs[51] = monitor_0 if monitor_0 is not None else MONITOR_0
+        regs[52] = monitor_1 if monitor_1 is not None else MONITOR_1
+
+        return regs
+
+    @classmethod
+    def regRunSimple(cls, readback=True):
+        """
+        just run the thing
+        :return: register packet data
+        """
+        return cls.regRun(1, 0, 0, 0, readback=readback)
+
+    @classmethod
+    def regRunSram(cls, startAddr, endAddr, loop=True, blockDelay=0, sync=249):
+        raise NotImplementedError("register run SRAM not valid for jump table DACs")
+
+    @classmethod
+    def regIdle(cls, delay):
+        """
+        :param delay: start delay
+        :return: numpy array of register bytes for idle mode
+        :rtype: numpy.ndarray
+        """
+        regs = np.zeros(cls.REG_PACKET_LEN, dtype='<u1')
+        regs[0] = 0  # do not start, daisy pass-through
+        regs[1] = 0  # no readback
+        regs[43:45] = littleEndian(int(delay), 2)
+        regs[51] = 16
+        regs[52] = 10
+        return regs
+
+    @classmethod
+    def regDebug(cls, word1, word2, word3, word4):
+        raise NotImplementedError("Not sure what debug means for the JT")
+
+    # regClockPolarity -- OK
+    # regDebug -- OK
+    # regI2C -- OK
+    # makeSram -- OK
+
+    def load(self, jt, sram, page=None):
+        """ Get a load packet for this DAC.
+
+        A load packet is a packet to the direct ethernet server that has
+        commands for loading the jump table and the SRAM.
+
+        :param jump_table.JumpTable jt: jump table, from make_jump_table
+        :param sram: sram data
+        :param page: None (anything else is invalid for JT boards)
+        :return: packet to the direct ethernet server
+        """
+        if page is not None:
+            raise NotImplementedError("page argument not valid for jump table")
+        p = self.makePacket()
+        p.write(jt.toString())
+        self.makeSRAM(sram, p)
+        return p
+
+    @classmethod
+    def make_jump_table(cls, entries, counters=None, start_address_ns=0):
+        """ Make a jump table out of the given entries and counters.
+
+        Note that each entry is a tuple of (string, list[int]). The string
+        is the entry name (END, IDLE, etc), and the integers are the arguments
+        for that entry. Note that from/to addresses are given in ns here, and
+        must be divisible by four. See
+        ghz_fpga_server.FPGAServer.jump_table_add_entry for further details.
+
+        :param list[(str, list[int])] entries: JT entries
+        :param list[int] counters: counter values, or None for all 0s
+        :param int start_address_ns: SRAM start address, in ns
+        :return: jump table object
+        :rtype: jump_table.JumpTable
+        """
+        jt_entries = []
+
+        for name, arg in entries:
+            if name == 'CHECK':
```

Comment:
I agree, but I came to the conclusion that ultimately we'd have to have this `case`-logic somewhere unless we changed the FPGA server's method from "add jump table entry" to "add jt end" + "add jt idle" + ...

In the future we can figure out a better way to do this.